### PR TITLE
Fix excessive padding on CarPlay during active navigation.

### PR DESCRIPTION
**Pull request by MaximAlien**
Closing #3604.

Since there is no ability to detect whether CarPlay head-unit is in either left of right driving side mode (at least I was not able to find it, feel free to provide any suggestions if you have any) I had to use `safeAreaInsets` value as a criteria.

### CODE REVIEW

**Review by ShanMa1991 — APPROVED** [review]

**Comment by ShanMa1991** [comment]
It's great if we could add this to `changeLog`.

File: Sources/MapboxNavigation/CarPlayNavigationViewController.swift [hunk 1]
```diff
@@ -395,6 +408,40 @@ open class CarPlayNavigationViewController: UIViewController {
...
+    public override func viewSafeAreaInsetsDidChange() {
+        super.viewSafeAreaInsetsDidChange()
+        
+        // Trigger update of view constraints to correctly position views like `SpeedLimitView` and
+        // `CarPlayCompassView`.
+        view.setNeedsUpdateConstraints()
+    }
+    
+    public override func updateViewConstraints() {
+        // Since there is no ability to detect current driving side mode of the CarPlay head-unit,
```
Comment (1ec5) [inline Sources/MapboxNavigation/CarPlayNavigationViewController.swift:420]: Does `UIApplication.shared.userInterfaceLayoutDirection` reflect the driving side? CarPlay ostensibly implements the driving side adaptation feature using RTL language support.

Comment (MaximAlien) [reply Sources/MapboxNavigation/CarPlayNavigationViewController.swift:420]: No matter what driving side I choose in CarPlay head-unit options I always get `.leftToRight` in `UIUserInterfaceLayoutDirection`. It's interesting that `Sony XAV-AX1000` doesn't even provide changing driving side. With `Intellidash Pro` it's possible, so I suspect that head-unit support this option only by drawing certain menus on the right.

Comment (1ec5) [reply Sources/MapboxNavigation/CarPlayNavigationViewController.swift:420]: (For posterity, it does not.)

**Pull request merged**